### PR TITLE
[LangRef] Spell out alias attribute/metadata violations are UB.

### PR DESCRIPTION
**Pull request by fhahn**
Update the documentation for the noalias attribute, !alias.scope and !loop.parallel_accesses metadata to clarify they are UB on voilation the noalias property.

### CODE REVIEW

**Review by nunoplopes — APPROVED** [review]
LGTM

File: llvm/docs/LangRef.rst [hunk 1]
```diff
@@ -6936,9 +6938,9 @@ does not carry useful data and need not be preserved.
 noalias memory-access sets. This means that some collection of memory access
 instructions (loads, stores, memory-accessing calls, etc.) that carry
 ``noalias`` metadata can specifically be specified not to alias with some other
-collection of memory access instructions that carry ``alias.scope`` metadata.
-Each type of metadata specifies a list of scopes where each scope has an id and
-a domain.
+collection of memory access instructions that carry ``alias.scope`` metadata. If
+accesses from different collections alias, the behaivor is undefined. Each type
```
Comment (nikic) [inline llvm/docs/LangRef.rst:6942]:
```suggestion
accesses from different collections alias, the behavior is undefined. Each type
```

**Comment by nikic** [comment]
Trying to remind myself why exactly using "load is poison, store is UB" semantics wouldn't work:

If we have
```
store v2, p
v = load p
```
where we have claimed that these are noalias, then the `store` is well-defined  and the `load` returns poison (due to conflicting store).

If we now move the load above the store (because they don't alias)
```
v = load p
store v2, p
```
then `load` returns a value and `store` is UB due to conflicting prior store. So we went from poison to UB.

So I believe that for scoped alias metadata only UB semantics can work, as specified here. Presumably a similar argument generalizes to the other metadata.

File: llvm/docs/LangRef.rst (hunk 1 above)
Comment (fhahn) [reply llvm/docs/LangRef.rst:6942]: Should be fixed, thanks!

**Review by nikic — APPROVED** [review]
LGTM, but please wait a bit in case there are more comments.

**Pull request merged**